module/os/linux/zfs/zpl_super.c

```diff
@@ -380,7 +389,11 @@ zpl_prune_sb(uint64_t nr_to_scan, void *arg)
 
 const struct super_operations zpl_super_operations = {
 	.alloc_inode		= zpl_inode_alloc,
+#ifdef HAVE_INODE_FREE
+	.free_inode		= zpl_inode_free,
+#else
 	.destroy_inode		= zpl_inode_destroy,
+#endif
```

Comment:
I believe we still want to register `.destroy_inode` even when `.free_inode` is available.  It's just that the RCU-delayed parts have been moved in to `.free_inode`.

Reply:
Well I tried to adhere to the note in `porting.rst` to the fullest extent possible, as it doesn't even say that .free_inode is now required if one doesn't want to run into unexplainable behavior (perhaps only with some kernel releases along the way to full folio conversion); I was wondering what more in terms of problems can we expect if I try to go against that note, in a sense of leaving `.destroy_inode` as it was.

Do you see any consequences of merging `destroy_inode` into `evict_inode` from the ZFS point of view? I couldn't think of anything so that's why I decided to do away with `destroy_inode` by merging it, just an abundance of caution, aka "what else aren't they telling me in the note, when they didn't even say it's mandatory but just recommended"

Reply:
I mean, if the inode gets as far as `evict_inode` being called on it, it's already done and `zfs_zget` is going to keep looping until it disappears completely, isn't it so?